### ISSUE TITLE
dockerng.volume_present fails when no volumes already exist on the system

### ISSUE THREAD

**Issue by bberberov** [saltstack/salt#32493]
The dockerng state module cannot create volumes when no volumes exist on the host.

### Setup
I have one master and one minion on the same host.

Use the example from https://docs.saltstack.com/en/latest/ref/states/all/salt.states.dockerng.html#salt.states.dockerng.volume_present as `test.sls`

`/test.sls`
```salt
volume_foo:
  dockerng.volume_present
```

### Steps to Reproduce Issue
1. Remove all existing docker volumes

2. Run the state:
  ```bash
  salt '*' state.sls test
  ```
  ```cmd
  ----------
            ID: volume_foo
      Function: dockerng.volume_present
        Result: False
       Comment: An exception occurred in this state: Traceback (most recent call last):
                  File "/usr/lib/python2.7/dist-packages/salt/state.py", line 1626, in call
                    **cdata['kwargs'])
                  File "/usr/lib/python2.7/dist-packages/salt/loader.py", line 1492, in wrapper
                    return f(*args, **kwargs)
                  File "/usr/lib/python2.7/dist-packages/salt/states/dockerng.py", line 2204, in volume_present
                    volumes = [v for v in __salt__['dockerng.volumes']()['Volumes'] if v['Name'] == name]
                TypeError: 'NoneType' object is not iterable
       Started: 16:10:28.322448
      Duration: 3.183 ms
       Changes:
```

3. Create a simple volume
  ```cmd
  docker volume create --name fix
  ```

4. Run the state again:
  ```bash
  salt '*' state.sls test
  ```
  ```cmd
  ----------
            ID: volume_foo
      Function: dockerng.volume_present
        Result: True
       Comment: 
       Started: 16:17:29.783824
      Duration: 4.826 ms
       Changes:   
                ----------
                created:
                    ----------
                    Driver:
                        local
                    Mountpoint:
                        /srv/store465g/docker/volumes/volume_foo/_data
                    Name:
                        volume_foo
  ```

### Versions Report
```cmd
# salt --versions-report
Salt Version:
           Salt: 2015.8.8.2
 
Dependency Versions:
         Jinja2: 2.7.2
       M2Crypto: Not Installed
           Mako: 0.9.1
         PyYAML: 3.10
          PyZMQ: 14.0.1
         Python: 2.7.6 (default, Jun 22 2015, 17:58:13)
           RAET: Not Installed
        Tornado: 4.2.1
            ZMQ: 4.0.4
           cffi: Not Installed
       cherrypy: Not Installed
       dateutil: 2.4.2
          gitdb: 0.5.4
      gitpython: 0.3.2 RC1
          ioflo: Not Installed
        libgit2: Not Installed
        libnacl: Not Installed
   msgpack-pure: Not Installed
 msgpack-python: 0.3.0
   mysql-python: 1.2.3
      pycparser: Not Installed
       pycrypto: 2.6.1
         pygit2: Not Installed
   python-gnupg: Not Installed
          smmap: 0.8.2
        timelib: Not Installed
 
System Versions:
           dist: Ubuntu 14.04 trusty
        machine: x86_64
        release: 4.4.0-13-generic
         system: Ubuntu 14.04 trusty
```

```cmd
# docker version
Client:
 Version:      1.10.3
 API version:  1.22
 Go version:   go1.5.3
 Git commit:   20f81dd
 Built:        Thu Mar 10 15:54:52 2016
 OS/Arch:      linux/amd64

Server:
 Version:      1.10.3
 API version:  1.22
 Go version:   go1.5.3
 Git commit:   20f81dd
 Built:        Thu Mar 10 15:54:52 2016
 OS/Arch:      linux/amd64
```

```cmd
# pip show docker-py
---
Name: docker-py
Version: 1.7.2
Location: /usr/local/lib/python2.7/dist-packages
Requires: requests, six, websocket-client
```

**Comment by jfindlay** [comment]
@bberberov, thanks for reporting.

**Comment by rallytime** [comment]
@bberberov Please see the fix in #32561.

**Comment by bberberov** [comment]
Works for me in salt 2016.3.1 (Boron)
